backend/src/chats/role/role.controller.ts

```diff
@@ -0,0 +1,42 @@
+import {
+	Controller,
+	Get,
+	Post,
+	Body,
+	Patch,
+	Param,
+	Delete,
+} from '@nestjs/common';
+import { RoleService } from './role.service';
+import { CreateRoleDto } from './dto/create-role.dto';
+import { UpdateRoleDto } from './dto/update-role.dto';
+
+@Controller('roles')
+export class RoleController {
+	constructor(private readonly roleService: RoleService) {}
+
+	@Post()
+	create(@Body() createRoleDto: CreateRoleDto) {
+		return this.roleService.create(createRoleDto);
+	}
+
+	@Get()
+	findAll() {
+		return this.roleService.findAll();
```

Comment:
Do we wanna add relations for the findAll method as well?

Reply:
the relationship are devined in the chatuser_role table, which is visible for you if you check the tables